Weapons with explosions are incompatible with WorldGuard

### Current Behavior

Weapons that produce explosions call their own EntityExplodeEvent but do it with an unmodifiable List. When WorldGuard is installed, it tries to manipulate this list (due to a region protection) causing an exception.

### Expected Behavior

WorldGuard should be able to modify the List passed to the event resulting in certain blocks damaged by the explosion to be protected by the region.

### Steps To Reproduce

1. Use an explosive weapon of any kind
2. Shoot the projectile towards a WorldGuard region

### Version and Log

```
[Server thread/ERROR]: Could not pass event EntityExplodeEvent to WorldGuard v7.0.7+216b061
org.bukkit.event.EventException: null
    at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot-api-1.19.3-R0.1-SNAPSHOT.jar:?]
    at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-api-1.19.3-R0.1-SNAPSHOT.jar:?]
    at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-api-1.19.3-R0.1-SNAPSHOT.jar:?]
    at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-api-1.19.3-R0.1-SNAPSHOT.jar:?]
    at me.deecaad.weaponmechanics.weapon.explode.Explosion.explode(Explosion.java:253) ~[?:?]
    at me.deecaad.weaponmechanics.weapon.explode.Explosion$1.run(Explosion.java:194) ~[?:?]
    at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftTask.run(CraftTask.java:82) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:415) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1249) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:388) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1201) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1017) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:301) ~[spigot-1.19.3-R0.1-SNAPSHOT.jar:3670-Spigot-454acb7-bd29f41]
    at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.UnsupportedOperationException
    at java.util.ImmutableCollections.uoe(ImmutableCollections.java:142) ~[?:?]
    at java.util.ImmutableCollections$AbstractImmutableCollection.clear(ImmutableCollections.java:149) ~[?:?]
    at com.sk89q.worldguard.bukkit.listener.WorldGuardEntityListener.onEntityExplode(WorldGuardEntityListener.java:530) ~[?:?]
    at jdk.internal.reflect.GeneratedMethodAccessor400.invoke(Unknown Source) ~[?:?]
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
    at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
    at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-api-1.19.3-R0.1-SNAPSHOT.jar:?]
    ... 13 more
```

### Anything else?

_No response_

--- a/WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/weapon/explode/BlockDamage.java
+++ b/WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/weapon/explode/BlockDamage.java
@@ -21,6 +21,7 @@ import org.bukkit.inventory.ItemStack;
 import org.jetbrains.annotations.NotNull;
 
 import javax.annotation.Nullable;
+import java.util.stream.Collectors;
 import java.util.*;
 
 public class BlockDamage implements Serializer<BlockDamage> {
@@ -142,7 +143,7 @@ public class BlockDamage implements Serializer<BlockDamage> {
     }
 
     public List<Block> filterBreakbleBlocks(List<Block> blocks) {
-        return blocks.stream().filter(block -> getBreakMode(block.getType()) == BreakMode.BREAK).toList();
+        return blocks.stream().filter(block -> getBreakMode(block.getType()) == BreakMode.BREAK).collect(Collectors.toList());
     }
 
     public int getDurability(Block block) {

